src/components/TokenSelection/TokenImportRow.tsx

```diff
@@ -0,0 +1,64 @@
+import classNames from "classnames";
+import { TokenInfo } from "@uniswap/token-lists";
+import { useTranslation } from "react-i18next";
+
+export type TokenImportRowProps = {
+  /**
+   * TokenInfo object
+   */
+  token: TokenInfo;
+  /**
+   * onClick event, either setSignerToken or setSenderToken
+   */
+  onClick: (val: string) => void;
+  /**
+   * Whether counterpart token is already selected; Still allows user to select that token
+   */
+  selected?: boolean;
+  /**
+   * Whether current token is already selected; Prevents token row click
+   */
+  disabled?: boolean;
+};
+
+const TokenImportRow = ({
+  token,
+  onClick,
+  selected,
+  disabled,
+}: TokenImportRowProps) => {
+  const { t } = useTranslation(["balances", "common"]);
+
+  return (
+    <div
+      className={classNames(
+        "grid items-center grid-flow-col hover:bg-gray-200 dark:hover:bg-gray-700 cursor-pointer",
+        selected && "opacity-40",
+        disabled && "opacity-40 pointer-events-none"
+      )}
+      style={{
+        gridTemplateColumns: "auto minmax(auto, 1fr) auto minmax(0, 72px)",
+        gridGap: "16px",
+      }}
+    >
+      <img
+        src={token.logoURI || "https://via.placeholder.com/150"}
+        className="w-6"
+        alt="hello"
+      />
+      <div className="flex flex-col justify-start">
+        <h3 className="flex flex-col">{token.symbol}</h3>
+        <h3 className="text-gray-400">{token.name}</h3>
+      </div>
+      <span></span>
+      <button
```

Comment:
`<button disabled />`

Give button a disabled prop and there is no need to check the disabled component prop.

Reply:
One point for discussion here: TokenImportRow has no onclick events when clicking on the row itself, so should it even be wrapped as a button? For uniformity sake, I guess it should.

Also there shouldn't actually be a need for disabled or selected prop for importing tokenImportRow, since the only purpose it serves is to import the token.

Reply:
Ah yes I see now. I didn't notice that there is no onClick event listener. Also according to W3C recommendation you shouldn't wrap a button in a button. So in this case you should keep it as div. My bad!